src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckBomEditCollector.java

```diff
@@ -185,10 +183,9 @@ public BlackDuckBomEditCollector(final JsonExtractor jsonExtractor, final BlackD
         try {
             final LinkableItem componentItem = new LinkableItem(BlackDuckContent.LABEL_COMPONENT_NAME, versionBomComponent.getComponentName(), versionBomComponent.getComponent());
             Optional<LinkableItem> componentVersionItem = createComponentVersionItem(versionBomComponent);
-            final List<PolicyRuleView> policyRules = getBlackDuckService().getAllResponses(versionBomComponent, VersionBomComponentView.POLICY_RULES_LINK_RESPONSE);
-            for (final PolicyRuleView rule : policyRules) {
-                final Optional<PolicySummaryStatusType> policySummaryStatusTypeFromRule = getPolicySummaryStatusTypeFromRule(rule);
-                if (policySummaryStatusTypeFromRule.isPresent() && !PolicySummaryStatusType.IN_VIOLATION.equals(policySummaryStatusTypeFromRule.get())) {
+            final List<VersionBomPolicyRuleView> policyRules = getBlackDuckService().getAllResponses(versionBomComponent, VersionBomComponentView.POLICY_RULES_LINK_RESPONSE);
+            for (final VersionBomPolicyRuleView rule : policyRules) {
+                if (!PolicySummaryStatusType.IN_VIOLATION.equals(rule.getPolicyApprovalStatus())) {
```

Comment:
Will this result in a NullPointerException if the PolicyApprovalStatus is null?

Reply:
Since it is the argument of equals, it shouldn't.